app/forms/post_form.rb

```diff
@@ -0,0 +1,56 @@
+class PostForm
+  include ActiveModel::Model
+  include Virtus.model
+  extend CarrierWave::Mount
+
+  validates :title,   presence: true, length: { maximum: 50 }
+  validates :content, presence: true, length: { maximum: 10000 }
+
+  attribute :user_id,     Integer
+  attribute :category_id, Integer
+  attribute :title,       String
+  attribute :content,     String
+  attribute :status,      Boolean
+
+  mount_uploader :picture, PictureUploader
+
+  attr_accessor :pictures
+
+  def initialize(post = Post.new)
+    @post = post
+    self.attributes = @post.attributes if @post.persisted?
+  end
+
+  def assign_attributes(params = {})
+    @params = params
+    # 画像情報からインスタンスを生成する
+    attributes = params.to_h[:pictures_attributes]
+    @pictures ||= []
+    attributes&.map do |attribute|
+      picture = Picture.new(attribute)
+      @pictures.push(picture)
+    end
+    # 画像以外の情報でpostを更新する
+    @params.delete(:pictures_attributes)
+    @post.assign_attributes(params) if @post.persisted?
+    super(params)
+  end
+
+  def save
+    return false if invalid?
+    if @post.persisted?
+      @post.pictures = pictrues unless pictures.nil?
```

Comment:
28行目で`@pictures ||= []`してるので、nilには絶対になりません。blank?にしてください。
もっというと、if ~~ present? の方が読み易いと思います。
ただ、空配列を入れてもエラーにはならない気がするので、今回の問題とは関係なさそうです。
```suggestion
      @post.pictures = pictrues if pictures.present?
```

Reply:
ここは完全に勘違いしていました。
DMでもこのような表記をしていたので、そちらもまた修正しておきます。